drivers/starburst/src/metabase/driver/implementation/execute.clj

```diff
@@ -121,11 +121,12 @@
 
 (defmethod sql-jdbc.execute/read-column-thunk [:starburst Types/TIMESTAMP_WITH_TIMEZONE]
   [_ ^ResultSet rset _ ^long i]
-  ;; Converts TIMESTAMP_WITH_TIMEZONE to instant, then to OffsetDateTime with default time zone.
+  ;; Converts TIMESTAMP_WITH_TIMEZONE to java.time.ZonedDateTime with default time zone.
   (fn []
-    (when-let [t (.getObject rset i java.sql.Timestamp)] 
-      (let [instant (.toInstant t)] 
-        (.atOffset instant (t/zone-offset 0))))))
+    (let [zonedDateTime (.getObject rset i java.time.ZonedDateTime)
+          utcTimeZone (java.time.ZoneId/of "UTC")]
+      (.toOffsetDateTime (.withZoneSameInstant zonedDateTime utcTimeZone))
+    )))
```

Comment:
so is the idea here is we are creating a zonedDateTime, which means we shouldn't lose any info about the time zone, and then we are setting the time zone to UTC, then converting to offsetDateTime? 

Is this related to fixing the [bug](https://github.com/starburstdata/metabase-driver/issues/58) here? 

Did this end up fixing the issue? Can you add some testing details to the PR overview that shows it is fixed?

Reply:
The bug is actually not a bug. It's PostgreSQL which exhibits some weird behavior. But we confirmed with Metabase that the Starburst driver has the right behavior.

This PR is a refactor which should not affect the behavior.